jbpm-wb-kie-server/jbpm-wb-kie-server-backend/src/test/java/org/jbpm/workbench/ks/integration/KieServerIntegrationClientsTest.java

```diff
@@ -93,83 +103,106 @@ public void testServerTemplateClients() {
 
-        assertEquals(1,
-                     kieServerIntegration.getServerTemplatesClients().size());
-        assertServerInstanceFailedEndpoint(kieServerIntegration,
-                                           serverInstance1);
-        assertContainerFailedEndpoint(kieServerIntegration,
-                                      serverInstance1,
-                                      containerSpec.getId());
+        assertEquals(1, kieServerIntegration.getServerTemplatesClients().size());
+        assertServerInstanceFailedEndpoint(kieServerIntegration, serverInstance1);
+        assertContainerFailedEndpoint(kieServerIntegration, serverInstance1, containerSpec.getId());
 
-        assertEquals(1,
-                     kieServerIntegration.getServerInstancesById().size());
+        assertEquals(1, kieServerIntegration.getServerInstancesById().size());
         assertNull(kieServerIntegration.getServerInstancesById().get(serverInstanceId1));
         assertNotNull(kieServerIntegration.getServerInstancesById().get(serverInstanceId2));
 
         kieServerIntegration.onServerInstanceDisconnected(new ServerInstanceDisconnected(serverInstanceId2));
 
-        assertEquals(1,
-                     kieServerIntegration.getServerTemplatesClients().size());
-        assertServerInstanceFailedEndpoint(kieServerIntegration,
-                                           serverInstance2);
-        assertContainerFailedEndpoint(kieServerIntegration,
-                                      serverInstance2,
-                                      containerSpec.getId());
+        assertEquals(1, kieServerIntegration.getServerTemplatesClients().size());
+        assertServerInstanceFailedEndpoint(kieServerIntegration, serverInstance2);
+        assertContainerFailedEndpoint(kieServerIntegration, serverInstance2, containerSpec.getId());
 
-        assertEquals(0,
-                     kieServerIntegration.getServerInstancesById().size());
+        assertEquals(0, kieServerIntegration.getServerInstancesById().size());
         assertNull(kieServerIntegration.getServerInstancesById().get(serverInstanceId1));
         assertNull(kieServerIntegration.getServerInstancesById().get(serverInstanceId2));
 
         kieServerIntegration.onServerTemplateDeleted(new ServerTemplateDeleted(serverTemplateId));
-        assertEquals(0,
-                     kieServerIntegration.getServerTemplatesClients().size());
+        assertEquals(0, kieServerIntegration.getServerTemplatesClients().size());
+    }
+
+    @Test
+    public void testCreateAvailableClients() throws Exception {
+        final String serverTemplateId = "kie-server-test";
+        final String serverInstanceId1 = serverTemplateId + "@1";
+        final String serverInstanceId2 = serverTemplateId + "@2";
+        final ContainerSpec containerSpec = newContainerSpec();
+        final ServerInstance serverInstance1 = new ServerInstance();
```

Comment:
1. Please group the four lines dealing with serverInstance1 initialization into single block (like you did for serverInstance2 below).
2. The serverInstance1 is not being used anywhere after initialization, is that intended? (serverInstance2 is being added to serverTemplate after init ...)

Reply:
fixed